midend/expr_uses.h

```diff
@@ -21,19 +21,52 @@ limitations under the License.
 
 /* Should this be a method on IR::Expression? */
 
+/// Functor to check if an expression uses an lvalue.  The lvalue is specified as a
+/// a cstring, which can be the name of a variable with optional field names and constant
+/// array indexes for fields of headers or structs or unions or elements of stacks.
```

Comment:
Am I wrong in thinking that we should be able to use the ReferenceMap to determine this stuff? (Perhaps with some additional munging for array indices.)

Reply:
In theory, yes, but it would be more complicated, and it would depend on the refMap being up to date.

That latter thing is the big problem with the refMap (and typeMap) in general -- it means you can't use them reliably in Transform passes (only in Inspectors) because as soon as you transform anything, they're invalidated which may corrupt later transformations in the same pass.  Since we use this in LocalCopyProp and LocalCopyProp is a (complex) transform, it hits this problem in a big way.

This is also why LocalCopyProp uses strings to name lvalues rather than `IR::Expression *`s -- the latter can change when things change (and there may be multiple copies of them) so using them as map keys is a non-starter.